[docs] Actions Refactoring (#4844)
DevExpress/testcafe: docs/articles/documentation/reference/test-api/testcontroller/getnativedialoghistory.md

---
layout: docs
title: t.getNativeDialogHistory Method
permalink: /documentation/reference/test-api/testcontroller/getnativedialoghistory.html
---
# t.getNativeDialogHistory Method

Provides a history of the invoked native dialogs.

```text
t.getNativeDialogHistory() → Promise<[{type, text, url}]>
```

> `t.getNativeDialogHistory` returns a stack of history entries (i.e., an array in which the latest dialog has an index of `0`).

Reviewer: ```suggestion
`t.getNativeDialogHistory` returns a stack of history entries (that is, an array in which the latest dialog has an index of `0`).
```

> Each entry corresponds to a certain native dialog that appears in the main window or in an `<iframe>`.

Reviewer: ```suggestion
Each entry corresponds to a native dialog that appears in the main window or in an `<iframe>`.
```
